extensions/default/src/DicomWebDataSource/retrieveStudyMetadata.js

```diff
@@ -35,8 +35,9 @@ export function retrieveStudyMetadata(
   }
 
   // Already waiting on result? Return cached promise
-  if (StudyMetaDataPromises.has(StudyInstanceUID)) {
-    return StudyMetaDataPromises.get(StudyInstanceUID);
+  const studyRetrieveId = dicomWebClient.name + StudyInstanceUID;
+  if (StudyMetaDataPromises.has(studyRetrieveId)) {
```

Comment:
I think I would be more comfortable using interpolation for creating the `studyRetrieveId` and also maybe adding a separator like a slash for debugging purposes. Something like...

```
const studyRetrieveId = `${dicomWebClient.name}/${StudyInstanceUID}`;
```